pyproject.toml

```diff
@@ -77,6 +77,40 @@ fmt = "black ."
 lint = "ruff ."
 test = "pytest . --cov anywidget --cov-report term-missing"
 
+# https://github.com/charliermarsh/ruff
+[tool.ruff]
+line-length = 88
+target-version = "py37"
+src = ["anywidget", "tests"]
+extend-select = [
+    "E",    # style errors
+    "F",    # flakes
+    "D",    # pydocstyle
+    "I",    # isort
+    "UP",   # pyupgrade
+    "S",    # bandit
+    "C4",   # flake8-comprehensions
+    "B",    # flake8-bugbear
+    "A001", # flake8-builtins
+    "TID",  # flake8-tidy-imports
+    "RUF",  # ruff-specific rules
+]
```

Comment:
Generally I'm all for less config, stronger defaults, but it seems to get the most out of ruff requires lots of opt-in. Is that correct? I'm familiary with `E`, `F`, `D`, `I` and now `UP` but have no idea what the rest of these are so I trust you.

Reply:
Yes, unfortunately ruff doesn't turn on most of the good stuff by default.  The ones you're familiar with are the big ones (I think isort in particular is important) but I think ruff only includes E and F by default.  RUF is nice since it removed unused type ignores.
The remainder are honesty not that important, I'm actually inclined to remove them here in the interest of less config (they'll rarely hit).

Reply:
Ok, in that case let's do `E`, `F`, `D`, `I`, `UP`, and `RUF` :)

Reply:
changed